src/tests/system/tests/test_sssctl.py

```diff
@@ -269,3 +273,681 @@ def test_sssctl__check_misplaced_option(client: Client):
+    assert "Attribute 'ldap_ri' is not allowed in section 'domain/ldap2'" in res.stdout, "Wrong error message"
+
+
+@pytest.mark.tools
+@pytest.mark.topology(KnownTopology.Client)
+def test_sssctl__config_does_not_exist(client: Client):
+    """
+    :title: sssctl detects missing config
+    :setup:
+        1. Start SSSD, so default config is autimatically created
+        2. Remove config
+    :steps:
+        1. Call sssctl config-check
+        2. Check error message
+    :expectedresults:
+        1. config-check detects an error
+        2. Error message is properly set
+    :customerscenario: False
+    """
+    client.sssd.common.local()
+    client.sssd.start()
+    client.fs.rm("/etc/sssd/sssd.conf")
+
+    result = client.sssctl.config_check()
+    assert result.rc != 0, "Config-check did not detect misconfigured config"
+    assert "File /etc/sssd/sssd.conf does not exist" in result.stdout, "Wrong error message on stdout"
+
+
+@pytest.mark.tools
+@pytest.mark.ticket(bz=1677994)
+@pytest.mark.topology(KnownTopology.Client)
+def test_sssctl__check_ldap_host_object_class_in_domain(client: Client):
+    """
+    :title: sssctl config-check allow ldap_host_object_class in domain section
+    :setup:
+        1. Add ldap_host_object_class to local domain section
+        2. Start SSSD
+    :steps:
+        1. Call sssctl config-check
+    :expectedresults:
+        1. config-check succeed
+    :customerscenario: True
+    """
+    client.sssd.default_domain = "local"
+    client.sssd.common.local()
+    client.sssd.domain["ldap_host_object_class"] = "ipService"
+    client.sssd.start(check_config=False)
+
+    result = client.sssctl.config_check()
+    assert result.rc == 0, "Config-check failed"
+
+
+@pytest.mark.tools
+@pytest.mark.ticket(bz=1677994)
+@pytest.mark.topology(KnownTopology.Client)
+def test_sssctl__check_ldap_host_object_class_in_sssd(client: Client):
+    """
+    :title: sssctl config-check do not allow ldap_host_object_class in sssd section
+    :setup:
+        1. Add ldap_host_object_class to sssd section
+        2. Start SSSD
+    :steps:
+        1. Call sssctl config-check
+    :expectedresults:
+        1. config-check succeed
+    :customerscenario: True
+    """
+    client.sssd.common.local()
+    client.sssd.sssd["ldap_host_object_class"] = "ipService"
+    client.sssd.start(check_config=False)
+
+    result = client.sssctl.config_check()
+    assert result.rc != 0, "Config-check did not detect misconfigured config"
+    assert (
+        "Attribute 'ldap_host_object_class' is not allowed in section 'sssd'" in result.stdout
+    ), "Wrong error message on stdout"
+
+
+@pytest.mark.tools
+@pytest.mark.ticket(bz=1856861)
+@pytest.mark.topology(KnownTopology.Client)
+def test_sssctl__enabling_2FA(client: Client):
+    """
+    :title: False warnings are logged in sssd.log file after enabling 2FA prompting
```

Comment:
I didn't check the old test but should we be asserting that there are no warnings in sssd.log ? 

From the BZ

```
Steps to Reproduce:
1. Enable 2FA prompting settings in sssd.conf
2. Restart sssd service
3. Check /var/log/sssd/sssd.log file


Actual results:
False errors/warnings are observed in /var/log/sssd/sssd.log file after restarting sssd service post enabling 2FA prompting settings in sssd.conf.
---
(Tue Jul 14 11:01:17:259149 2020) [sssd] [sss_ini_call_validators] (0x0020): [rule/allowed_sections]: Section [prompting/password] is not allowed. Check for typos.
(Tue Jul 14 11:01:17:259335 2020) [sssd] [sss_ini_call_validators] (0x0020): [rule/allowed_sections]: Section [prompting/2fa] is not allowed. Check for typos.
---


Expected results:
There shouldn't be any warnings/errors in /var/log/sssd/sssd.log
```

Reply:
I precisely converted the test and did not look at the bz. This is the original test:
```
    @pytest.mark.tier1
    @pytest.mark.converted('test_sssctl.py', 'test_sssctl__enabling_2FA')
    def test_0025_check2FA(self, multihost, backupsssdconf):
        """
        :title: IDM-SSSD-TC: Configuration validation: Check false
         warnings are logged in sssd.log file after enabling 2FA prompting
         settings in sssd.conf
        :id: 3a1060db-0120-4270-b669-aae8923613a0
        :customerscenario: True
        """
        # Automation of BZ1856861
        tools = sssdTools(multihost.client[0])
        section1 = "prompting/2fa/sshd"
        domain_params1 = {'first_prompt': 'Enter OTP Token Value:',
                          'single_prompt': 'single_prompt'}
        section2 = "prompting/2fa"
        domain_params2 = {'single_prompt': 'True',
                          'first_prompt': 'Prompt1',
                          'second_prompt': 'Prompt2'}
        tools.sssd_conf(section1, domain_params1)
        tools.sssd_conf(section2, domain_params2)
        multihost.client[0].service_sssd('restart')
        sssctl_cmd = 'sssctl config-check'
        cmd = multihost.client[0].run_command(sssctl_cmd, raiseonerr=False)
        assert cmd.returncode == 0
```

Reply:
But i think this way of testing is similar to checking the logs. Because if these messages were generated:
```
(Tue Jul 14 11:01:17:259149 2020) [sssd] [sss_ini_call_validators] (0x0020): [rule/allowed_sections]: Section [prompting/password] is not allowed. Check for typos.
(Tue Jul 14 11:01:17:259335 2020) [sssd] [sss_ini_call_validators] (0x0020): [rule/allowed_sections]: Section [prompting/2fa] is not allowed. Check for typos.
```
They would be generated by sssctl config-check and that is what I am checking